src/phoenix/server/api/types/Project.py

```diff
@@ -226,76 +226,71 @@ def document_evaluation_names(
         )
 
     @strawberry.field
-    def trace_evaluation_summary(
+    async def trace_evaluation_summary(
         self,
+        info: Info[Context, None],
         evaluation_name: str,
         time_range: Optional[TimeRange] = UNSET,
     ) -> Optional[EvaluationSummary]:
-        project = self.project
-        eval_trace_ids = project.get_trace_evaluation_trace_ids(evaluation_name)
-        if not eval_trace_ids:
-            return None
-        trace_ids = project.get_trace_ids(
-            start_time=time_range.start if time_range else None,
-            stop_time=time_range.end if time_range else None,
-            trace_ids=eval_trace_ids,
+        base_query = (
+            select(models.TraceAnnotation)
+            .join(models.Trace)
+            .where(models.Trace.project_rowid == self.id_attr)
+            .where(models.SpanAnnotation.annotator_kind == "LLM")
+            .where(models.SpanAnnotation.name == evaluation_name)
         )
-        evaluations = tuple(
-            evaluation
-            for trace_id in trace_ids
-            if (
-                evaluation := project.get_trace_evaluation(
-                    trace_id,
-                    evaluation_name,
+        unfiltered = base_query
+        filtered = base_query
+        if time_range:
+            filtered = filtered.where(
+                and_(
+                    time_range.start <= models.Span.start_time,
+                    models.Span.start_time < time_range.end,
                 )
             )
-            is not None
-        )
+
+        # todo: implement filter condition
+        async with info.context.db() as session:
+            evaluations = list(await session.scalars(filtered))
+            all_evaluations = list(await session.scalars(unfiltered))
         if not evaluations:
             return None
```

Comment:
We can exit early, and the second query doesn't need to over-fetch.
```suggestion
            evaluations = await session.scalars(filtered)
            if not evaluations:
                return None
            labels = await session.scalars(unfiltered.with_only_columns(distinct(models.TraceAnnotation.label))
```